packages/runtime/src/helpers/analysis.ts

```diff
@@ -87,18 +83,25 @@ let hasWarnedAboutNextVersion = false
 /**
  * Uses Next's swc static analysis to extract the config values from a file.
  */
-export const extractConfigFromFile = async (apiFilePath: string): Promise<ApiConfig> => {
+export const extractConfigFromFile = async (apiFilePath: string, appDir: string): Promise<ApiConfig> => {
   if (!apiFilePath || !existsSync(apiFilePath)) {
     return {}
   }
 
   try {
     if (!extractConstValue) {
-      extractConstValue = require('next/dist/build/analysis/extract-const-value')
+      // eslint-disable-next-line import/no-dynamic-require
+      extractConstValue = require(findModuleFromBase({
+        paths: [appDir],
+        candidates: ['next/dist/build/analysis/extract-const-value'],
+      }) ?? 'next/dist/build/analysis/extract-const-value')
```

Comment:
This is a bit messy - especially part after `??` - it's done because `findModuleFromBase` can return `null`, and calling `require(null)` won't result in `MODULE_NOT_FOUND` error that we handle below - so the reason for `??` part is to actually produce above error in case module can't be resolved

Reply:
[sand] you could potentially move the try/catch from `findModuleFromBase` to `getServerFile` (which is the only place that `findModuleFromBase` is used) and then you don't have to the juggling with `??`

Reply:
I decided to do https://github.com/netlify/next-runtime/pull/2059/commits/04f3164c275d50ac12f5958936b9ffe2f4c46d27 instead - I looked at how throwing would look like in `findModuleFromBase` and realized that removing `try/catch` there would result in throwing in first missed candidate, and to make it somewhat work it would be quite messy with storing exceptions and rethrowing them when we completed iterating over candidates and still didn't find one.

Also thinking more - this would change semantics of `findModuleFromBase` which is not end of the world, but I prefer to avoid changing shared utils behavior if it can be addressed in single place that would benefit from it (and possibly cause problems in all the other places)